packages/flutter_markdown/lib/src/markdown_raw.dart

```diff
@@ -0,0 +1,400 @@
+          syntaxHighlighter: syntaxHighlighter
+        )
+      )
+    );
+  }
+
+  MarkdownBodyRaw createMarkdownBody({
+    String data,
+    MarkdownStyleRaw markdownStyle,
+    SyntaxHighlighter syntaxHighlighter
+  }) {
+    return new MarkdownBodyRaw(
+      data: data,
+      markdownStyle: markdownStyle,
+      syntaxHighlighter: syntaxHighlighter
+    );
+  }
+}
+
+/// A [Widget] that renders markdown formatted text. It supports all standard
+/// markdowns from the original markdown specification found here:
+/// https://daringfireball.net/projects/markdown/ This class doesn't implement
+/// any scrolling behavior, if you want scrolling either wrap the widget in
+/// a [ScrollableViewport] or use the [MarkdownRaw] widget.
+class MarkdownBodyRaw extends StatefulComponent {
+
+  /// Creates a new Markdown [Widget] that renders the markdown formatted string
+  /// passed in as [data]. You need to pass in a [markdownStyle] that defines
+  /// how the code is rendered. Code blocks are by default not using syntax
+  /// highlighting, but it's possible to pass in a custom [syntaxHighlighter].
+  ///
+  /// Typically, you may want to wrap the [MarkdownBodyRaw] widget in a
+  /// [Padding] and a [ScrollableViewport], or use the [Markdown class]
+  ///
+  ///     new ScrollableViewport(
+  ///       child: new Padding(
+  ///         padding: new EdgeDims.all(16.0),
+  ///         child: new MarkdownBodyRaw(
+  ///           data: markdownSource,
+  ///           markdownStyle: myStyle
+  ///         )
+  ///       )
+  ///     )
+  MarkdownBodyRaw({
+    this.data,
+    this.markdownStyle,
+    this.syntaxHighlighter
+  });
+
+  /// Markdown styled text
+  final String data;
+
+  /// Style used for rendering the markdown
+  final MarkdownStyleRaw markdownStyle;
+
+  /// The syntax highlighter used to color text in code blocks
+  final SyntaxHighlighter syntaxHighlighter;
+
+  _MarkdownState createState() => new _MarkdownState();
+
+  MarkdownStyleRaw createDefaultStyle(BuildContext context) {
+    return null;
+  }
+}
+
+class _MarkdownState extends State<MarkdownBodyRaw> {
+
+  void initState() {
+    super.initState();
+
+    MarkdownStyleRaw markdownStyle = config.markdownStyle ?? config.createDefaultStyle(context);
+    SyntaxHighlighter syntaxHighlighter = config.syntaxHighlighter ?? new _DefaultSyntaxHighlighter(markdownStyle.code);
+
+    _cachedBlocks = _blocksFromMarkup(config.data, markdownStyle, syntaxHighlighter);
+  }
+
+  List<_Block> _cachedBlocks;
+
+  Widget build(BuildContext context) {
+    List<Widget> blocks = <Widget>[];
+    for (_Block block in _cachedBlocks) {
+      blocks.add(block.build(context));
+    }
+
+    return new Column(
+      alignItems: FlexAlignItems.stretch,
+      children: blocks
+    );
+  }
+}
+
+List<_Block> _blocksFromMarkup(String data, MarkdownStyleRaw markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+  List<String> lines = data.replaceAll('\r\n', '\n').split('\n');
```

Comment:
Please add a TODO comment about how this is inefficient and we should replace this with an implementation that doesn't malloc a temporary buffer the size of the data.